substratest/assets.py

```diff
@@ -264,7 +264,7 @@ class Objective(_Asset):
 
 
 @dataclasses.dataclass(frozen=True)
-class TupleDataset(_DataclassLoader):
+class TesttupleDataset(_DataclassLoader):
```

Comment:
minor and not 100% sure it would improve readability: but a TesttupleDataset is a TraintupleDataset with an additional perf field.

So we could have:
```python
class TesttupleDataset(TraintupleDataset):
    perf: float
```
```

Reply:
And we could keep the `TupleDataset` that is common for train like tuples and a dedicated `TesttupleDataset` inheriting from the `TupleDataset`.

Reply:
Since inheriting assets wasn't done anywhere I went with the duplication. I'll make the change.

Reply:
It's already done for Algo for instance.

Reply:
One other quick solution would be to make the perf field optional by adding a `= None` suffix.

Reply:
To recap our discussion in the office: the names `(Train/Test)tupleDataset` are ok because TraintupleDataset in used for Traintuples and CompositeTraintuples only.

Using Inheritance failed because of the __annotiations__ that are not enriched with the parent's class __annotations__.

In the end we went with the existing solution and will refactor this later on.